### ISSUE TITLE
Quit org-agenda will kill the last buffer

### ISSUE THREAD

**Issue by dsdshcym** [syl20bnr/spacemacs#4256]
I don't know if this is a personal issue. But after the commit which evilified org-agenda
buffer. I ran into this issue:

> When I hit `q` to quit org-agenda, the last buffer in which I opened org-agenda
will be killed too.

**Comment by TheBB** [comment]
I can reproduce. Might be a bug with `org-agenda-restore-windows-after-quit`.

**Comment by dsdshcym** [comment]
@TheBB And I also need to hit `q` twice in order to quit org-agenda.

**Comment by TheBB** [comment]
That one I can't confirm.

**Comment by dsdshcym** [comment]
@TheBB It seems that when I hit `q` for the first time, the last buffer will be
killed. And the second time, the org-agenda buffer is killed.

**Comment by nixmaniack** [comment]
I can confirm this too. Do you have sticky agenda? I tried debugging this but couldn't find exact cause. 
This is what is working for me sanely now. 
```elisp
(setq org-agenda-sticky t)
(setq org-agenda-window-setup 'current-window)
(setq org-agenda-restore-windows-after-quit nil)
```

**Comment by dsdshcym** [comment]
@nixmaniack 
Yeah. This is a working work-around. But I really want to restore windows after I quit org-agenda.

**Comment by dsdshcym** [comment]
@TheBB It seems that this bug was fixed by org-20151228. 

Can you check again on your side?

**Comment by TheBB** [comment]
Yes, I can't reproduce with the latest version.
